### ISSUE TITLE
Support DNS-01 Challenge

### ISSUE THREAD

**Issue by coolaj86** [Daplie/le-acme-core#9]
This may already be supported implicitly. I think everything that needs to be exposed is and the option is now included as of https://github.com/Daplie/letiny-core/pull/8/commits/e7a36123ca4d9d2c72d255fa2ed02acff74ebbf7#diff-8237c7da8c30d998035487db609e6f9dR368

* [ ] Learn about the DNS-01 spec
* [ ] Test if it works by setting `keyAuthorization` to some DNS record? I guess?

**Comment by philkunz** [comment]
@coolja86 Awesome. :) Yes, take a look here: https://gitlab.com/pushrocks/cert/blob/master/ts/cert.hook.ts
So you are 100% right! It just requires setting the challenge as txt record with an `_acme-challenge.` prefix. I think for many scenarios this is much easier than hacking into the reverse proxy or whatever handles your traffic to host the challenge over http.

**Comment by coolaj86** [comment]
Hey @philkunz! All you need to do is implement an `le-challenge` strategy and plug in your code to update the DNS records instead of using the default.

You'll want to use v2 of [node-letsencrypt](https://github.com/Daplie/node-letsencrypt)

Here are some examples (all short and single file):

* [le-challenge-standalone](https://github.com/Daplie/le-challenge-memory)
* [le-challenge-interactive](https://github.com/Daplie/le-challenge-manual)
* [le-challenge-webroot](https://github.com/Daplie/le-challenge-webroot)

Basically there are five methods to implement:

* getOptions()
* set(opts, domain, token, key)
* get(opts, domain, token)
* remove(opts, domain, token)

I'm going to close this out, but please continue the discussion and I'll be happy to help answer further questions.

**Comment by coolaj86** [comment]
Just looked at your code, you've pretty much nailed it. Just wrap that into the format needed by `node-letsencrypt` and you're golden.

**Comment by philkunz** [comment]
@coolaj86 Perfect. Thank you so much. I'm looking into this this weekend and post here once I've migrated to node-letsencrypt from letsencrypt.sh

**Comment by philkunz** [comment]
@coolaj86 Any clue why this fails to retrieve a valid account?: https://gitlab.com/pushrocks/cert/blob/master/ts/cert.classes.letsencrypt.ts
